File: backend/src/main/java/com/inventory/repository/ProductRepository.java
package com.inventory.repository;

import com.inventory.entity.Product;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProductRepository extends JpaRepository<Product, Long> {
    
    @Query("SELECT p FROM Product p LEFT JOIN FETCH p.productMateriaPrimas prm LEFT JOIN FETCH prm.materiaPrima")
    List<Product> findAllWithMateriaPrimas();
    
    @Query("SELECT p FROM Product p LEFT JOIN FETCH p.productMateriaPrimas prm LEFT JOIN FETCH prm.materiaPrima WHERE p.id = :id")
    Product findByIdWithMateriaPrimas(Long id);
}

File: backend/src/main/kotlin/com/inventory/repository/ProductRepository.kt
package com.inventory.repository

import com.inventory.entity.Product
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface ProductRepository : JpaRepository<Product, Long> {
    
    @Query("SELECT p FROM Product p LEFT JOIN FETCH p.productRawMaterials prm LEFT JOIN FETCH prm.rawMaterial")
    fun findAllWithRawMaterials(): List<Product>
    
    @Query("SELECT p FROM Product p LEFT JOIN FETCH p.productRawMaterials prm LEFT JOIN FETCH prm.rawMaterial WHERE p.id = :id")
    fun findByIdWithRawMaterials(id: Long): Product?
}
